medmodels/data_synthesis/mtgan/builder.pyi

```diff
@@ -0,0 +1,53 @@
+from pathlib import Path
+from typing import Dict
+
+from typing_extensions import Unpack
+
+from medmodels.data_synthesis.mtgan.model.gan import TrainingHyperparameters
+from medmodels.data_synthesis.mtgan.modules.postprocessor import (
+    AttributeType,
+    PostprocessingHyperparameters,
+)
+from medmodels.data_synthesis.mtgan.modules.preprocessor import (
+    PreprocessingHyperparameters,
+)
+from medmodels.data_synthesis.mtgan.mtgan import MTGAN
+from medmodels.data_synthesis.mtgan.mtgan_model import MTGANModel
+from medmodels.medrecord.types import Group, MedRecordAttribute
+
+class MTGANBuilder:
+    seed: int
+
+    patients_group: Group
+    concepts_group: Group
+    time_attribute: MedRecordAttribute
+
+    training_hyperparameters: TrainingHyperparameters
+    preprocessing_hyperparameters: PreprocessingHyperparameters
+    postprocessing_hyperparameters: PostprocessingHyperparameters
+
+    training_hyperparameters_json: TrainingHyperparameters
+    preprocessing_hyperparameters_json: PreprocessingHyperparameters
+    postprocessing_hyperparameters_json: PostprocessingHyperparameters
```

Comment:
I've thought a bit about this and I think that we should stick to only one set of hyperparameters

```suggestion
    training_hyperparameters: TrainingHyperparameters
    preprocessing_hyperparameters: PreprocessingHyperparameters
    postprocessing_hyperparameters: PostprocessingHyperparameters
```

because currently we're simply assuming that the user wants the normal hyperparameters to overrule the json hyperparameters (without explicitly telling the user that we are assuming this).
I think we should keep it simple and include in the docstrings that the order of operations matters, so the "newer" operation that sets any hyperparameters might overrule an "older" operation (.e.g `{...}.with_preprocessor_hyperparameters(...).load_hyperparameters_from(...)` vs. `{...}.load_hyperparameters_from(...).with_preprocessor_hyperparameters(...)` )